src/events.js

```diff
@@ -1462,6 +1456,7 @@ const addNewBox = async (userName) => {
                     }
                 }
             }
+            document.getElementById('shippingModalChooseBox').setAttribute('data-new-box', newBoxId);
```

Comment:
I like your approach of adding an attribute that assigns the New Box ID state of the modal select dropdown Element. When the new box is created the set attribute "data-new-box" gets cleared.